### PR TITLE
test: fix flaky force_recovery test

### PR DESCRIPTION
**Pull request by EvgenyMekhanik**
In the test, there is a place, where it was checked that the amount
of valid data in snapshot in case when it was truncated is less than
in case we write garbage to it. Often it's really so, but depends on
the place of truncation/garbage location. Removed this check, because
on different systems, snapshot size is slightly different each time
you run test, so check will not pass every time.
Follow-up #5422

### CODE REVIEW

**Review by Korablev77 — APPROVED** [review]
LGTM

**Comment by Korablev77** [comment]
Oh wait, are you sure it is really so?
> on different systems, snapshot size is slightly different

**Comment by EvgenyMekhanik** [comment]
Also fix following error:
Static buffer to save snapshot filename, reused later in `xlog_cursor_open`
function. So when we log this name after it we get corrupted name that has
nothing to do with the real name. I save this name in buffer located on the
function stack and use it later.

File: src/box/memtx_engine.c [hunk 1]
```diff
@@ -161,6 +161,13 @@ memtx_engine_recover_snapshot(struct memtx_engine *memtx,
 	int64_t signature = vclock_sum(vclock);
 	const char *filename = xdir_format_filename(&memtx->snap_dir,
 						    signature, NONE);
+	/*
+	 * Filename saved in static buffer, which reused later, so we need to
+	 * save filename in storage, which is local for this function.
+	 */
+	char name[PATH_MAX];
+	strncpy(name, filename, PATH_MAX);
+	name[PATH_MAX - 1] = '\0';
```
Comment (Gerold103) [inline src/box/memtx_engine.c:170]: You can use `strlcpy()`. It is defined here: https://github.com/tarantool/tarantool/blob/master/src/trivia/util.h#L466.

Comment (EvgenyMekhanik) [reply src/box/memtx_engine.c]: Thanks! Fixed

File: src/box/memtx_engine.c [hunk 2]
```diff
@@ -205,9 +211,9 @@ memtx_engine_recover_snapshot(struct memtx_engine *memtx,
 	 */
 	if (!xlog_cursor_is_eof(&cursor)) {
 		if (!memtx->force_recovery)
-			panic("snapshot `%s' has no EOF marker", filename);
```
Comment (Korablev77) [inline src/box/memtx_engine.c:208]: If we reach so far, why not use cursor->name instead of filename? xlog_cursor_close() even has comment saying that it is ok to address to cursor's fields even after closing it.

File: test/box/gh-5422-broken_snapshot.test.lua [hunk 3]
```diff
@@ -49,7 +49,7 @@ end;
 test_run:cmd("setopt delimiter ''");
 
 garbage_size = 500
-corruption_offset = 15000
+corruption_offset = 5000
```
Comment (Korablev77) [inline test/box/gh-5422-broken_snapshot.test.lua:52]: Why did you change offset? In commit message you are saying that we are about only to remove one check...
